src/create-client.ts

```diff
@@ -63,24 +63,25 @@ export async function createClient(
   let _authkitClientState: State = "INITIAL";
 
   const _needsRefresh = () => {
-    if (_authkitClientState === "AUTHENTICATED") {
-      const accessToken = memoryStorage.getItem(storageKeys.accessToken) as
-        | string
-        | undefined;
-      const expiresAt = memoryStorage.getItem(storageKeys.expiresAt) as
-        | number
-        | undefined;
-
-      if (!accessToken || !expiresAt) {
-        return true;
-      }
+    if (_authkitClientState !== "AUTHENTICATED") {
```

Comment:
i think there's a potential issue here still where you could trigger a bunch of refresh requests by calling `getAccessToken` several times in the window where a refresh is needed... i'll address that separately though. (should behave fine other than we will issue more requests than is necessary).